.github/workflows/reader-api-ci.yml

```diff
@@ -0,0 +1,48 @@
+name: Reader API CI
+
+on:
+  push:
+    branches: [ main, develop ]
+    paths:
+      - 'reader-api/**'
+      - '.github/workflows/reader-api-ci.yml'
+  pull_request:
+    branches: [ main, develop ]
+    paths:
+      - 'reader-api/**'
+      - '.github/workflows/reader-api-ci.yml'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Set up JDK 21
+      uses: actions/setup-java@v4
+      with:
+        java-version: '21'
+        distribution: 'temurin'
+        
+    - name: Cache Maven dependencies
+      uses: actions/cache@v4
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+        
+    - name: Build and test reader-api with coverage
```

Comment:
We're missing a few important environment variables:

QUARKUS_LANGCHAIN4J_AZURE_OPENAI_RESOURCE_NAME
QUARKUS_LANGCHAIN4J_AZURE_OPENAI_DEPLOYMENT_NAME
QUARKUS_LANGCHAIN4J_AZURE_OPENAI_API_KEY

Please pull the values for these environment variables from the corresponding secrets.